src/ApproximateTNContraction/ITensors.jl

```diff
@@ -0,0 +1,49 @@
+#
+# ITensors.jl extensions
+#
+
+# Generalize siteind to n-dimensional lattice
+function ITensors.siteind(st::SiteType, N1::Integer, N2::Integer, Ns::Integer...; kwargs...)
+  s = siteind(st; kwargs...)
+  if !isnothing(s)
+    ts = "n1=$N1,n2=$N2"
+    for i in eachindex(Ns)
+      ts *= ",n$(i + 2)=$(Ns[i])"
+    end
+    return addtags(s, ts)
+  end
+  return isnothing(s) && error(space_error_message(st))
+end
+
+# Generalize siteinds to n-dimensional lattice
+function ITensors.siteinds(
```

Comment:
Hopefully this can be replace by the constructors of the form `siteinds("S=1/2", named_grid((4, 4)))`?